src/lib/components/Modal/Modal.jsx

```diff
@@ -1,52 +1,46 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import { createPortal } from 'react-dom';
-import {
-  RUIContext,
-  withGlobalProps,
-} from '../../provider';
+import { withGlobalProps } from '../../provider';
 import { classNames } from '../../utils/classNames';
-import { transferProps } from '../_helpers/transferProps';
-import {
-  Toolbar,
-  ToolbarItem,
-} from '../Toolbar';
-import Button from '../Button';
-import ScrollView from '../ScrollView';
 import styles from './Modal.scss';
 
 export class Modal extends React.Component {
   constructor(props) {
     super(props);
 
     this.childrenWrapperRef = React.createRef();
-    this.submitButtonRef = React.createRef();
 
     this.keyPressHandler = this.keyPressHandler.bind(this);
   }
 
   componentDidMount() {
-    const { autoFocus } = this.props;
+    const {
+      autoFocus,
+      primaryButtonRef,
+    } = this.props;
 
     window.document.addEventListener('keydown', this.keyPressHandler, false);
 
     // If `autoFocus` is set to `true`, following code finds first form field element
-    // (input, textarea or select) or submit button and auto focuses it. This is necessary
+    // (input, textarea or select) or primary button and auto focuses it. This is necessary
     // to have focus on one of those elements to be able to submit form by pressing Enter key.
-    if (autoFocus && this.childrenWrapperRef && this.childrenWrapperRef.current) {
-      const childrenWrapperElement = this.childrenWrapperRef.current;
-      const childrenElements = childrenWrapperElement.querySelectorAll('*');
-      const formFieldEl = Array.from(childrenElements).find(
-        (element) => ['INPUT', 'TEXTAREA', 'SELECT'].includes(element.nodeName),
-      );
+    if (autoFocus) {
+      if (this.childrenWrapperRef?.current != null) {
+        const childrenWrapperElement = this.childrenWrapperRef.current;
+        const childrenElements = childrenWrapperElement.querySelectorAll('*');
```

Comment:
I think we can use one the `React.Children.*` methods here (https://reactjs.org/docs/react-api.html#reactchildren).
probably `toArray()` or `map()`.

That would allow us to get rid of the ref.

Reply:
Unfortunately not. Generally, I don't want to touch code which functionality is proven and tested, but that is not primary reason. Primary reason is that we have to detect HTML element, not React element. Second reason is that It would require recursive calling as inputs can be place deeper in DOM tree, while querySelectorAll work like a charm and according to some articles its performance in not that bad.

Reply:
OK, makes sense